packages/connection/src/class-rest-connector.php

```diff
@@ -51,4 +104,100 @@ public function verify_registration( \WP_REST_Request $request ) {
 
 		return $this->connection->handle_registration( $registration_data );
 	}
+
+	/**
+	 * Handles verification that a site is registered
+	 *
+	 * @since 5.4.0
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return array|wp-error
+	 */
+	public static function remote_authorize( $request ) {
+		$xmlrpc_server = new Jetpack_XMLRPC_Server();
+		$result        = $xmlrpc_server->remote_authorize( $request );
+
+		if ( is_a( $result, 'IXR_Error' ) ) {
+			$result = new WP_Error( $result->code, $result->message );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Get connection status for this Jetpack site.
+	 *
+	 * @since 4.3.0
+	 *
+	 * @return WP_REST_Response Connection information.
+	 */
+	public static function connection_status() {
+		$status     = new Status();
+		$connection = new Manager();
+
+		return rest_ensure_response(
+			array(
+				'isActive'     => $connection->is_active(),
+				'isStaging'    => $status->is_staging_site(),
+				'isRegistered' => $connection->is_registered(),
+				'devMode'      => array(
+					'isActive' => $status->is_development_mode(),
+					'constant' => defined( 'JETPACK_DEV_DEBUG' ) && JETPACK_DEV_DEBUG,
+					'url'      => site_url() && false === strpos( site_url(), '.' ),
+					'filter'   => apply_filters( 'jetpack_development_mode', false ),
+				),
+			)
+		);
+	}
+
+
+	/**
+	 * Get plugins connected to the Jetpack.
+	 *
+	 * @since 8.6.0
+	 *
+	 * @return WP_REST_Response|WP_Error Response or error object, depending on the request result.
+	 */
+	public function get_connection_plugins() {
+		$plugins = $this->connection->get_connected_plugins();
+
+		if ( is_wp_error( $plugins ) ) {
+			return $plugins;
+		}
+
+		array_walk(
+			$plugins,
+			function( &$data, $slug ) {
+				$data['slug'] = $slug;
+			}
+		);
+
+		return rest_ensure_response( array_values( $plugins ) );
+	}
+
+	/**
+	 * Verify that user can view Jetpack admin page and can activate plugins.
+	 *
+	 * @since 8.8.0
+	 *
+	 * @return bool|WP_Error Whether user has the capability 'jetpack_admin_page' and 'activate_plugins'.
+	 */
+	public static function activate_plugins_permission_check() {
+		if ( current_user_can( 'jetpack_admin_page' ) && current_user_can( 'activate_plugins' ) ) {
```

Comment:
The `jetpack_admin_page` capability is [set up in the Jetpack class](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L1099), so it won't be available when the `jetpack-connection` package is used without Jetpack. I wonder if we could write a new permission check for the `connection/plugins` endpoint instead of using `activate_plugins_permission_check()`. 

Alternatively, we could move the `jetpack_admin_page` capability to this package, but it seems like it's specific to Jetpack and should stay in the Jetpack class.

Sorry that I didn't notice this before!

Reply:
Hi @kbrown9, good catch!
Do you think we need that capability in there at all? It's set for everybody who has access to the dashboard, and the access is restricted by `activate_plugins` capability anyway.

I removed it, and reverted that part of code in the `class.core-rest-api-endpoints.php` to not affect other endpoints: https://github.com/Automattic/jetpack/pull/16384/commits/99390a2075edcadd9b5b8ee4615821208c48e3d7

Please let me know if I'm missing something here.

Reply:
Your changes look good to me - thanks!